src/LendingPool.sol

```diff
@@ -376,46 +391,47 @@ contract LiquidityPool is ERC20, Owned {
     }
 
     /** 
-     * @notice Syncs interest payments to the Liquidity providers and the treasury.
+     * @notice Syncs interest payments to the Lending providers and the treasury.
      * @param assets The total amount of underlying assets to be paid out as interests.
      * @dev The weight of each Tranche determines the relative share yield (interest payments) that goes to its Liquidity providers
      * @dev The Shares for each Tranche are rounded up, if the treasury receives the remaining shares and will hence loose
      *      part of their yield due to rounding errors (neglectable small).
      */
-    function _syncInterestsToLiquidityPool(uint256 assets) internal {
+    function _syncInterestsToLendingPool(uint256 assets) internal {
         uint256 remainingAssets = assets;
 
         for (uint256 i; i < tranches.length; ) {
             uint256 trancheShare = assets.mulDivUp(weights[i], totalWeight);
-            _mint(tranches[i], trancheShare);
+            supplyBalances[tranches[i]] += trancheShare;
+            totalSupply += trancheShare;
             unchecked {
                 remainingAssets -= trancheShare;
                 ++i;
             }
         }
 
-        //Protocol fee
-        _mint(treasury, remainingAssets);
+        // Protocol fee, not added to the totalSupply because treasury fund are not used in lending.
+        protocolFee += remainingAssets;
```

Comment:
Why use a separate variable for protocolFee?
We can add the fee for the treasury to the mapping supplyBalances.

Reply:
Then we can also use the Withdraw function to claim profits for the treasury (now the treasury can never withdraw their funds).

Reply:
We can add it to supplyBalances, but it shouldn't be available in totalSupply since it is not part of the supply. And supplyBalances total should be represented in totalSupply. These params will be useful when calculating utilisation (totalBorrow / totalSupply ) and the fee distribution. Thats why i seperated the protocolFee, since it is something else

Reply:
And I don't think it is a problem the treasury fee is part of the Lending pool, it will be small anyway, Let's discuss it tomorrow in stand-up

Reply:
I think in Aave the fee is also collected in aTokens, and if Uniswap ever switches on the fee it is also in LP-tokens they mint the fee, so it is possible to keep it in the total supply, but I understand your point, let's discuss tomorrow